python/treelite/gtil/gtil.py

```diff
@@ -60,45 +101,73 @@ def predict(
         2D NumPy array, with which to run prediction
     nthread : :py:class:`int <python:int>`, optional
         Number of CPU cores to use in prediction. If <= 0, use all CPU cores.
-    pred_margin : bool, optional
-        Deprecated. Set predict_type="raw" instead.
-    predict_type : string, optional
-        One of the following:
-        * "default": Usual prediction method. Sum over trees and apply post-processing.
-        * "raw": Sum over trees, but don't apply post-processing; get raw margin scores
-                 instead.
-        * "leaf_id": Output one (integer) leaf ID per tree.
-        * "score_per_tree": Output one or more margin scores per tree.
 
     Returns
     -------
     prediction : :py:class:`numpy.ndarray` array
-        Prediction
+        Prediction output. Expected output dimensions:
+
+        - (num_row,) for regressors and binary classifiers
+        - (num_row, num_class) for multi-class classifiers
+
+    Notes
+    -----
+    Treelite assigns a unique integer ID for every node in the tree, including leaf
```

Comment:
This is great; it clarifies the expected behavior perfectly. The only thing we might add is that this means that predict_leaf output may differ from the training frameworks.

Reply:
Yes, users should call dump_as_json function to get the correct mapping for IDs.